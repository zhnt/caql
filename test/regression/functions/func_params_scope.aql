let global_x = 100

function test_scope(x, y) {
    print("param x = " + string(x))
    print("global_x = " + string(global_x))
    
    let local_z = x + y
    print("local_z = " + string(local_z))
    
    {
        let x = 999  // shadow parameter
        print("shadowed x = " + string(x))
        print("local_z still = " + string(local_z))
    }
    
    print("param x restored = " + string(x))
}

test_scope(50, 25)
