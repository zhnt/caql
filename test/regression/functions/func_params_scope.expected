param x = 50
global_x = 100
local_z = 75
shadowed x = 999
local_z still = 75
param x restored = 50
